rust/agents/validator/src/server/eigen_node.rs

```diff
@@ -0,0 +1,272 @@
+                status: ServiceStatus::Up,
+            },
+        ];
+        Json(services)
+    }
+
+    /// Method to return the health of a service
+    pub async fn service_health_handler(service_id: String) -> impl IntoResponse {
+        // TODO: implement logic to check if the service is healthy
+        // now just return 200
+        format!("Health check for service: {}", service_id);
+        StatusCode::OK
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::net::SocketAddr;
+
+    use super::*;
+    use axum::http::StatusCode;
+    use prometheus::Registry;
+
+    async fn setup_test_server() -> (reqwest::Client, SocketAddr, Arc<CoreMetrics>) {
+        let core_metrics =
+            Arc::new(CoreMetrics::new("dummy_validator", 37582, Registry::new()).unwrap());
+        // Initialize the Prometheus registry
+        core_metrics
+            .latest_checkpoint()
+            .with_label_values(&["validator_observed", "ethereum"])
+            .set(42);
+
+        let node_api = EigenNodeAPI::new(
+            HyperlaneDomain::new_test_domain("ethereum"),
+            Arc::clone(&core_metrics),
+        );
+        let app = node_api.router();
+
+        // Running the app in the background using a test server
+        let server =
+            axum::Server::bind(&"127.0.0.1:0".parse().unwrap()).serve(app.into_make_service());
+        let addr = server.local_addr();
+        tokio::spawn(server);
+
+        // Create a client
+        let client = reqwest::Client::new();
+
+        (client, addr, core_metrics)
+    }
+
+    #[tokio::test]
+    async fn test_eigen_node_api() {
+        let (client, addr, _) = setup_test_server().await;
+        let res = client
+            .get(format!("http://{}/node", addr))
+            .send()
+            .await
+            .expect("Failed to send request");
+
+        // Check that the response status is OK
+        assert_eq!(res.status(), StatusCode::OK);
+
+        // what to expect when you're expecting
+        let expected = NodeInfo {
+            node_name: "Hyperlane Validator".to_string(),
+            spec_version: "0.1.0".to_string(),
+            node_version: "0.1.0".to_string(),
+        };
+
+        // check the response body if needed
+        let json: NodeInfo = res.json().await.expect("Failed to parse json");
+        assert_eq!(json, expected);
+    }
+
+    #[tokio::test]
+    async fn test_eigen_node_health_api() {
+        let (client, addr, core_metrics) = setup_test_server().await;
+        let res = client
+            .get(format!("http://{}/node/health", addr))
+            .send()
+            .await
+            .expect("Failed to send request");
+        assert_eq!(res.status(), StatusCode::SERVICE_UNAVAILABLE);
+
+        // if signed_checkpoint - observed_checkpoint <= 10 return 206 - partially healthy
+        core_metrics
+            .latest_checkpoint()
+            .with_label_values(&["validator_processed", "ethereum"])
+            .set(34);
```

Comment:
nit: could store the magic numbers in variables with suggestive names. That would reduce the need for comments, e.g. `let partially_healthy_observed_checkpoint = 34;` or so

Reply:
fixed